### ISSUE TITLE
systemctl reports mounts from /proc/self/mountinfo as needing reload

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#5461]
```
$ systemctl status /mnt/tmp
Warning: mnt-tmp.mount change on disk. ...
 Loaded: loaded (/proc/self/mountinfo)
...
```
I guess we should filter `/proc/self/mountinfo` as a source. It'll always be "newer" than the mount.

**Comment by poettering** [comment]
I figure everything below /proc, /sys and friends should be considered like that.

**Comment by poettering** [comment]
Fix waiting in #5487
